### DISCUSSION TITLE
Table Widget is not updated

### DISCUSSION THREAD

**Discussion by zbadshah** [filamentphp/filament#6400] [Help]
I have a table widget, which does not show latest data unless I refresh browser. Is it possible to do within the widget or the action? I tried poll method and others that I found in the discussion but nothing seems to work.

Would highly appreciate a feedback.

Thanks

**Comment by danharrin** [comment]
They are Livewire components, so you can emit an event from another class which updates data and then register a listener on the widget to trigger `$refresh`

**Reply by zbadshah** [reply]
Thank you dan. 
Please refer to an example or tutorial so I can understand what you said. 

Thanks again.

**Reply by danharrin** [reply]
https://laravel-livewire.com/docs/2.x/events#from-component
https://laravel-livewire.com/docs/2.x/events#event-listeners

**Reply by zbadshah** [reply]
Thanks Dan,
I was using following for refreshing table contents as per documentation.
use Filament\Resources\Table;
 
public static function table(Table $table): Table
{
    return $table
        // ...
        ->poll('10s');
}

However, after your last message when I was studying doing same with Livewire I found the following:
        protected function getTablePollingInterval(): ?string
        {
            return '6s';
        }

I tried this and it worked like a charm. Perhaps in documentation you may need to add this function also.

Thanks again for your help.

**Reply by danharrin** [reply]
Hmm I wouldn't do polling, since you will be refreshing the table when you don't need to. Events allow you to be much more specific with when it gets refreshed

**Reply by zbadshah** [reply]
Ok. I understand. 

Now I will make it event based. Thanks.
